src/main/java/org/folio/acquisitions_tools/service/LoginService.java

```diff
@@ -0,0 +1,68 @@
+package org.folio.acquisitions_tools.service;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+@Service
+public class LoginService {
+  private final RestTemplate restTemplate = new RestTemplate();
+  private String token;
+
+  @Value("${folio.api.url}")
+  private String okapiUrl;
+
+  @Value("${folio.login.url}")
+  private String loginUrl;
+
+  @Value("${folio.api.tenant}")
+  private String tenant;
+
+  @Value("${folio.api.username}")
+  private String username;
+
+  @Value("${folio.api.password}")
+  private String password;
+
+  public String getToken() {
+    if (token == null) {
```

Comment:
better to not have statful fields, better to login each time when doing rest call to avoid potential errors

Reply:
refactored to use as method